crates/oxc_linter/src/lib.rs

```diff
@@ -90,6 +115,44 @@ impl Linter {
             }
         }
 
+        let adapter = LintAdapter { semantic: semantic.clone() };
+        let la = Arc::from(adapter);
+        for input_query in TRUSTFALL_RULES.iter() {
+            for data_item in execute_query(
+                &SCHEMA,
+                Arc::clone(&la),
+                input_query.query.as_str(),
+                input_query.args.clone(),
```

Comment:
Can you just borrow here instead of needing `as_str()` and `clone()`?

If not, I might have messed up something in the `execute_query()` definition...

Reply:
Works for query, not for args though, that's not a reference that's a pass by value treemap